apps/yggdrasil/test/yggdrasil/user_role.exs

```diff
@@ -0,0 +1,104 @@
+defmodule Yggdrasil.UserRoleTest do
+  use ExUnit.Case, async: false
+
+  alias Yggdrasil.{Repo, User, Role, UserRole, RoleResource, Resource, Permission}
+
+  @user %{username: "tester",
+          password: "password",
+          password_confirmation: "password"}
+
+  @test_perms [:foo_perm, :bar_perm, :baz_perm]
+  @test_resources [:foo_res, :bar_res, :baz_res]
+  @test_roles [
```

Comment:
I think with your suggestion about the names for permissions and resources this becomes a simpler task and can be smaller.

Because what was getting cumbersome was creating a valid role.